src/test/ui/issues/issue-58856-1.stderr

```diff
@@ -0,0 +1,30 @@
+error: expected one of `)`, `,`, or `:`, found `>`
+  --> $DIR/issue-58856-1.rs:3:14
+   |
+LL |     fn b(self>
+   |         -    ^
+   |         |    |
+   |         |    help: `)` may belong here
+   |         unclosed delimiter
+
+error: expected one of `->`, `where`, or `{`, found `>`
+  --> $DIR/issue-58856-1.rs:3:14
+   |
+LL |     fn b(self>
+   |              ^ expected one of `->`, `where`, or `{` here
+
+error: expected one of `->`, `async`, `const`, `crate`, `default`, `existential`, `extern`, `fn`, `pub`, `type`, `unsafe`, `where`, or `}`, found `>`
+  --> $DIR/issue-58856-1.rs:3:14
+   |
+LL |     fn b(self>
+   |              ^ expected one of 13 possible tokens here
+
```

Comment:
The two extra errors here are less than ideal, but this is something that can be ameliorated by keeping track of unexpected errors and elide the later ones based on the primary span.

Reply:
I think you can get rid of them by adding `{}` after `>`

Reply:
@jethrogb I can for this test, but I would like to get rid of them in a more general way to account for this kind of code in the wild.